PocketKit/Sources/Sync/RecentSavesWidgetStore.swift

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// swiftlint:disable multiline_arguments_brackets
+import SharedPocketKit
+import WidgetKit
+
+enum RecentSavesStoreError: Error {
+    case invalidStorage
+}
+
+/// A store that provides a list of recent saves
+public protocol RecentSavesStore {
+    var isLoggedIn: Bool { get }
+    func setLoggedIn(_ isLoggedIn: Bool)
+    var recentSaves: [SavedItemContent] { get }
+    func updateRecentSaves(_ items: [SavedItemContent]) throws
+}
+
+/// A concrete implementation of `RecentSavesStore` used for the recent saves widget
+public struct RecentSavesWidgetStore: RecentSavesStore {
+    private static let isLoggedInKey = "RecentSavesWidgetLoggedInKey"
+    private static let recentSavesKey = "RecentSavesWidgetKey"
+    private static let groupID = "group.com.ideashower.ReadItLaterPro"
+
+    private var defaults: UserDefaults
+
+    /// Current recent saves list
+    public var recentSaves: [SavedItemContent] {
+        guard let encodedSaves = defaults.object(forKey: Self.recentSavesKey) as? Data,
+                let saves = try? JSONDecoder().decode([SavedItemContent].self, from: encodedSaves) else {
+            return []
+        }
+        return saves
+    }
+
+    /// Current logged in status
+    public var isLoggedIn: Bool {
+        defaults.bool(forKey: Self.isLoggedInKey)
+    }
+
+    public init(userDefaults: UserDefaults) {
+        self.defaults = userDefaults
+    }
+
+    /// Update the recent saves list with the given list
+    /// - Parameter items: the given list
+    public func updateRecentSaves(_ items: [SavedItemContent]) throws {
+        let encodedList = try JSONEncoder().encode(items)
+        defaults.setValue(encodedList, forKey: Self.recentSavesKey)
+    }
+
+    /// Sets the logged in status
+    /// - Parameter isLoggedIn: the logged in status to set
+    public func setLoggedIn(_ isLoggedIn: Bool) {
+        defaults.setValue(isLoggedIn, forKey: Self.isLoggedInKey)
+    }
```

Comment:
I may be missing something, but do we clear out recent saves when a user is logged out? Or is it "indirectly" done once a user logs out via the main app, and the recent saves section reloads to empty (my assumption)?